src/main/java/org/testng/internal/ITestInvoker.java

```diff
@@ -35,6 +35,9 @@ FailureContext retryFailed(
 
   ITestResult registerSkippedTestResult(ITestNGMethod testMethod, long start, Throwable throwable);
```

Comment:
Is it possible to remove this one?

Reply:
@juherr - Sure we can. But there are invocations which will not involve a `ITestResult` object. If we remove this method then those calls would need to unnecessarily pass a spurious `null` object, which I was trying to avoid and have it taken care of internally.